lib/compat/experimental/blocks.php

```diff
@@ -52,6 +52,13 @@ function build_comment_query_vars_from_block( $block ) {
 			}
 		}
 
+		$order = ! empty( $block->context['comments/order'] ) ? $block->context['comments/order'] : null;
+		if ( empty( $order ) && get_option( 'comment_order' ) ) {
+			$comment_args['order'] = get_option( 'comment_order' );
+		} else {
```

Comment:
Nit: If we assign the initial value to `$comment_args['order']` instead of `$order` we should not need this part after `else`.